src/Contracts/Generator.php

```diff
@@ -9,4 +9,9 @@ public function generate(): string;
     public function setTemplate(string $template): self;
 
     public function setData(array $data): self;
+
+    /**
+     * Allows to set generator-specific context.
+     */
+    public function setGeneratorContext(mixed $context): self;
```

Comment:
kāpēc visur mixed? ja negribi specificēt konkrētu klasi, varbūt var interfeisu ieviest?

Reply:
Mixed jo, tur var padot jebko iekšā, šobrīd mums ir tikai twigtopdf bet vēlāk tur var būt sazin kas, citiem ģeneratoriem nevajag zināt header/footer, citiem varbūt svarīgi ir kaut kas cits, šis ļauj Tavam interfeisam būt fleksiblam attiecībā pret katra ģeneratora specific vajadzībām. Manis pēc varam ieviest GeneratorContextInterface, kas NEKO nenosaka, kā tikai to ka šo interfeisa implementāciju var padot Generator interfeisam lai uzstādītu kontekstu

Reply:
arī citi mierīgi izmanto interfeisus, kas neko nedara, kā vien definē kautkādas definīcijas. es būtu par tādu, nevis mixed